### PR TITLE
Adding Scheduled Tasks Collector

### PR DESCRIPTION
**Pull request by mousavian**
Adding a new collector called `scheduled_task` that can be enabled to provide metrics about registered tasks configured in the Windows Task Scheduler Service. ( #808 )

### CODE REVIEW

**Comment by mousavian** [comment]
Any idea why CI failed? Not sure why it's still trying to download `go-ole v1.2.1` while this PR suggests higher version in mod file.

**Review by breed808 — COMMENTED** [review]
This looks rather nice! I'll admit my knowledge of the Win32 API is lacking, so I'm not able to adequately review most of the `getScheduledTasks()` function. @carlpett could you take a look?
Documentation is also very good, thanks for putting in the effort for that :+1:

You're right about the CI, I've been able to compile this PR locally with no issues. We may need to re-trigger the CI build to see if it's a repeatable error.

File: collector/scheduled_task.go
Comment (carlpett) [inline collector/scheduled_task.go:168]: As `getScheduledTasks` is called on each scrape, do we need to run initialization every time, or would it suffice to do once? I'm somewhat concerned about interactions with how the WMI library uses OLE as well, and especially with calling `CoUninitialize`.

Comment (carlpett) [reply collector/scheduled_task.go:168]: Also, we need to consider concurrent scrapes.

File: collector/scheduled_task.go
Comment (carlpett) [inline collector/scheduled_task.go:206]: Is there a reason not to define these as normal free functions? Am I just missing some reuse of local vars?

File: collector/scheduled_task.go
Comment (carlpett) [inline collector/scheduled_task.go:179]: We call this due to the `defer` anyway, any reason you call it explicitly "again"?

File: collector/scheduled_task.go
Comment (carlpett) [inline collector/scheduled_task.go:185]: Same here, is this needed?

**Review by carlpett — COMMENTED** [review]
Very nice! I have some questions on the OLE stuff, but generally :+1:

File: collector/scheduled_task.go
Comment (carlpett) [inline collector/scheduled_task.go:261]: On a final re-read, I noticed the use of `Must`-functions - this means we'll panic and crash if there's any issues here, which we really do not want to.

**Review by carlpett — CHANGES_REQUESTED** [review]
This looks very nice to me now, except one thing I missed earlier. With that fixed we'll be good to go :)

File: collector/scheduled_task.go
Comment (mousavian) [reply collector/scheduled_task.go]: 💯 + ✔️

**Comment by mousavian** [comment]
I have rebased this... should be ready to merge if tests are passing :)

**Review by carlpett — APPROVED** [review]
Nice, thank you!

**Comment by breed808** [comment]
@Gerrit0284 this change hasn't yet made it into a new release yet; this was implemented after v0.18.1.
You can wait for a new release, or build the exporter from `master`.

**Pull request merged**
